var/spack/repos/builtin/packages/octopus/package.py

```diff
@@ -134,12 +228,99 @@ def install(self, spec, prefix):
         if (spec.satisfies('%apple-clang') or
                 spec.satisfies('%clang') or
                 spec.satisfies('%gcc')):
-            args.extend([
-                'FCFLAGS=-O2 -ffree-line-length-none'
-            ])
+            # In case of GCC version 10, we will have errors because of
+            # argument mismatching. Need to provide a flag to turn this into a
+            # warning and build sucessfully
+            if (spec.satisfies('%gcc@10:')):
+                args.extend([
+                    'FCFLAGS=-O2 -ffree-line-length-none '
+                    '-fallow-argument-mismatch -fallow-invalid-boz',
+                    'FFLAGS=-O2 -ffree-line-length-none '
+                    '-fallow-argument-mismatch -fallow-invalid-boz'])
+            else:
+                args.extend([
+                    'FCFLAGS=-O2 -ffree-line-length-none',
+                    'FFLAGS=-O2 -ffree-line-length-none'
+                ])
 
+        autoreconf('-i')
         configure(*args)
         make()
         # short tests take forever...
         # make('check-short')
         make('install')
+
+    def test(self):
+        self.smoke_tests()
+
+    @run_after("install")
+    def smoke_tests(self):
```

Comment:
I didn't get the pattern to use the `spack install --test=root` flag to run tests during install; thank you for pointing this out below. With this new knowledge, I do indeed not want to run the tests automatically during install. 

However, I would like to re-use the same testing code during install, and also when `spack test run` is used. To avoid duplication, I have created three functions:


```python
 @run_after('install')
    @on_package_attributes(run_tests=True)
    def smoke_tests_after_install(self):
        """Function stub to run tests after install if desired
        (for example through `spack install --test=root octopus`)
        """
        self.smoke_tests()

    def test(self):
        """Entry point for smoke tests run through `spack test run octopus`.
        """
        self.smoke_tests()

    def smoke_tests(self):
        """Actual smoke tests for Octopus."""
        #
```

I have tested this, and believe it works as expected.